dot/rpc/modules/chain_integration_test.go

```diff
@@ -0,0 +1,423 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+//go:build integration
+// +build integration
+
+package modules
+
+import (
+	"io/ioutil"
+	"math/big"
+	"path/filepath"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/dot/state"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/ChainSafe/gossamer/lib/runtime"
+	"github.com/ChainSafe/gossamer/lib/runtime/wasmer"
+	"github.com/ChainSafe/gossamer/lib/trie"
+	"github.com/ChainSafe/gossamer/lib/utils"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+
+	database "github.com/ChainSafe/chaindb"
+	rtstorage "github.com/ChainSafe/gossamer/lib/runtime/storage"
+	"github.com/stretchr/testify/require"
+)
+
+// test data
+var (
+	sampleBodyBytes = *types.NewBody([]types.Extrinsic{[]byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}})
+	// sampleBodyString is string conversion of sampleBodyBytes
+	sampleBodyString = []string{"0x2800010203040506070809"}
+)
+
+func TestChainGetHeader_Genesis(t *testing.T) {
+	state := newTestStateService(t)
+	svc := NewChainModule(state.Block)
+
+	header, err := state.Block.BestBlockHeader()
+	require.NoError(t, err)
+
+	di := types.NewDigestItem()
+	prd, err := types.NewBabeSecondaryPlainPreDigest(0, 1).ToPreRuntimeDigest()
+	require.NoError(t, err)
+	err = di.Set(*prd)
+	require.NoError(t, err)
+
+	d, err := scale.Marshal(di)
+	require.NoError(t, err)
+
+	expected := &ChainBlockHeaderResponse{
+		ParentHash:     header.ParentHash.String(),
+		Number:         common.BytesToHex(header.Number.Bytes()),
+		StateRoot:      header.StateRoot.String(),
+		ExtrinsicsRoot: header.ExtrinsicsRoot.String(),
+		Digest: ChainBlockHeaderDigest{
+			Logs: []string{common.BytesToHex(d)},
+		},
+	}
+
+	hash := state.Block.BestBlockHash()
+
+	res := &ChainBlockHeaderResponse{}
+	req := &ChainHashRequest{Bhash: &hash}
+
+	err = svc.GetHeader(nil, req, res)
+	require.NoError(t, err)
+	require.Equal(t, expected, res)
+}
+
+func TestChainGetHeader_Latest(t *testing.T) {
+	state := newTestStateService(t)
+	svc := NewChainModule(state.Block)
+
+	header, err := state.Block.BestBlockHeader()
+	require.NoError(t, err)
+
+	di := types.NewDigestItem()
```

Comment:
Just a general comment, I think it would be better to use full names like `digestItem`, `di` is just plain confusing. I even prefer `request` instead of `res` for example, but that might be just my opinion / full name exaggeration. If it would be me I would rename `err` to `error` and that would blow up the Go type system so yeah I'm also a full name maniac.

So all good here, just a thing to keep in mind for future code :+1: